module/VuFind/src/VuFind/Controller/Plugin/Holds.php

```diff
@@ -250,22 +255,57 @@ public function validateDates($startDate, $requiredBy, $enabledFormFields)
                     $result['requiredByTS'] = 0;
                 }
                 if ($result['requiredByTS'] < strtotime('today')) {
-                    $errors[] = 'hold_required_by_date_invalid';
+                    $result['errors'][] = 'hold_required_by_date_invalid';
                 }
             } catch (DateException $e) {
-                $errors[] = 'hold_required_by_date_invalid';
+                $result['errors'][] = 'hold_required_by_date_invalid';
             }
         }
 
-        if (!$errors
+        if (!$result['errors']
             && in_array('startDate', $enabledFormFields)
             && in_array('requiredByDate', $enabledFormFields)
             && $result['startDateTS'] > $result['requiredByTS']
         ) {
-            $errors[] = 'hold_required_by_date_before_start_date';
+            $result['errors'][] = 'hold_required_by_date_before_start_date';
+        }
+
+        return $result;
+    }
+
+    /**
+     * Check if the user-provided "frozen until" date is valid.
+     *
+     * Returns validated date and/or an array of validation errors if there are
+     * problems.
+     *
+     * @param string $frozenUntil     User-specified "frozen until" date
+     * @param array  $extraHoldFields Hold form fields enabled by
+     * configuration/driver
+     *
+     * @return array
+     */
+    public function validateFrozenUntil(?string $frozenUntil, array $extraHoldFields
+    ): array {
+        $result = [
+            'frozenUntilTS' => null,
+            'errors' => [],
+        ];
+        if (!in_array('frozenUntil', $extraHoldFields) || empty($frozenUntil)) {
+            return $result;
+        }
+
+        try {
+            $result['frozenUntilTS'] = $frozenUntil
```

Comment:
If $frozenUntil is a "falsey" value, won't we return because of the empty check above? Is there any reason to check its value again here?

Reply:
Right, simplified.